ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/TransactionReceipt.java

```diff
@@ -115,7 +144,9 @@ public void writeTo(final RLPOutput out) {
     out.writeLongScalar(cumulativeGasUsed);
     out.writeBytesValue(bloomFilter.getBytes());
     out.writeList(logs, Log::writeTo);
-
+    if (withRevertReason && revertReason.isPresent()) {
+      out.writeBytesValue(BytesValue.wrap(revertReason.get().getBytes(StandardCharsets.UTF_8)));
```

Comment:
If someone has asked for the reason, but it doesn't exist, should we include an empty  string?
I.e. Client requested the RevertReason explicitly, probably should give it to them (even if it doesn't exist).

Reply:
Don't think an empty string makes sense as an empty string is a valid revert reason. And returning a null for the revert reason as it does now makes more sense as there is no revert reason value to return.

Reply:
👍 `null` return value for a `null` revert reason.

In this context `null` and an empty string would have different meanings i.e. no revert message vs an empty revert message.